src/Weasel.Core/Migrator.cs

```diff
@@ -118,23 +118,23 @@ public async Task ReadTemplatesAsync(string directory, CancellationToken ct = de
     {
         if (autoCreate == AutoCreate.None)
         {
-            return;
+            return Task.CompletedTask;
         }
 
         if (migration.Difference == SchemaPatchDifference.None)
         {
-            return;
+            return Task.CompletedTask;
         }
 
         if (!migration.Deltas.Any())
         {
-            return;
+            return Task.CompletedTask;
         }
 
         migration.AssertPatchingIsValid(autoCreate);
 
         logger ??= new DefaultMigrationLogger();
-        await executeDelta(migration, conn, autoCreate, logger, ct).ConfigureAwait(false);
+        return executeDelta(migration, conn, autoCreate, logger, ct);
```

Comment:
Why is this synchronous now?

Reply:
It's not synchronous, I just removed async from the method definition and removed redundant await (see. also above replacement of `return;` with `Task.CompletedTask`.